src/object.cpp

```diff
@@ -73,6 +73,7 @@ Object::Object(SharedRealm r, Obj const& o)
 , m_object_schema(&*m_realm->schema().find(ObjectStore::object_type_for_table_name(o.get_table()->get_name())))
 , m_obj(o)
 {
+  REALM_ASSERT(!m_obj.get_table() || (m_realm->get_group() == _impl::TableFriend::get_parent_group(*m_obj.get_table())));
```

Comment:
This should use realm->read_group() rather than adding another function for the same thing.

Reply:
But read_group() appear to have quite some overhead....?

Reply:
When the realm is already in a read transaction it just does the checks we do on every single property access anyway.

Reply:
Ok, I changed it to use read_group(), but can't help notice that read_group() potentially will do a begin_read(). So calling read_group() at the point I'm proposing may change behavior, because the different timing will allow begin_read() to attach to a different version. Are you fine with that @tgoyne ?

Reply:
If read_group() starts a read transaction here then we're aborting the process anyway because this assertion is going to fail.